serving/samples/jib-app-java/README.md

```diff
@@ -0,0 +1,101 @@
+# Jib Sample App
+
+A sample app that demonstrates using
+[Jib](https://github.com/GoogleContainerTools/jib) to build Java applications
+and running with Knative Serving.
+
+Jib can package any Java application without needing a Dockerfile nor Docker
+installed.
+
+This builds the
+[Spring Boot Webflux](https://github.com/spring-projects/spring-boot/tree/v2.1.2.RELEASE/spring-boot-samples/spring-boot-sample-webflux)
+sample app, creates a container image, and deploys it into Knative using the
+[Knative Jib Build Template](https://github.com/knative/build-templates/tree/master/jib).
+
+## Prerequisites
+
+- [Install Knative Serving](../../../install/README.md)
+- [Configure Image Push Credentials](../../../build/auth.md#basic-authentication-docker)
+
+## Running
+
+This sample uses the
+[Jib build template](https://github.com/knative/build-templates/tree/master/jib)
+in the [build-templates](https://github.com/knative/build-templates/) repo.
+
+Install the Jib Maven build template:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/knative/build-templates/master/jib/jib-maven.yaml
+```
+
+Then you can deploy this to Knative Serving from the root directory by entering
+the following commands:
+
+```shell
+# Replace <your-project-here> with your own registry
+export REPO="gcr.io/<your-project-here>"
+
+perl -pi -e "s@DOCKER_REPO_OVERRIDE@$REPO@g" sample.yaml
+
+# Create the Kubernetes resources
+kubectl apply -f sample.yaml
+```
+
+Once deployed, you will see that it first build and revision. See the revision status.
+
+```shell
+kubectl describe revisions jib-sample-app-00001
+```
+
+Once the `BuildSucceed` status is `True`, then the build container image will be deployed.
+
+To access this service using `curl`, we first need to determine its ingress
+address:
+
+```shell
+# In Knative 0.2.x and prior versions, the `knative-ingressgateway` service was used instead of `istio-ingressgateway`.
+INGRESSGATEWAY=knative-ingressgateway
+
+# The use of `knative-ingressgateway` is deprecated in Knative v0.3.x.
+# Use `istio-ingressgateway` instead, since `knative-ingressgateway`
+# will be removed in Knative v0.4.
+if kubectl get configmap config-istio -n knative-serving &> /dev/null; then
+    INGRESSGATEWAY=istio-ingressgateway
```

Comment:
I'd just use this.  We're about to delete the other in 0.4, and 0.3 has been out for a while.

Reply:
updated.